packages/editor/src/lib/editor/sessions/Session.ts

```diff
@@ -0,0 +1,224 @@
+import { Editor } from '../Editor'
+import { TLEventInfo } from '../types/event-types'
+
+/**
+ * A simple event object that can be used to prevent the default behavior of a session event.
+ * @public */
+export class SessionEvent<T extends object = object> {
+	constructor(
+		public info: T,
+		public duration: number,
+		public elapsed: number
+	) {}
+	isDefaultPrevented = false
+	preventDefault() {
+		this.isDefaultPrevented = true
+	}
+}
+
+/**
+ *  A session info object that can be used to define event handlers for a session.
+ *  @public */
+export interface SessionEventHandlers<T extends object = object> {
+	onBeforeStart?: (event: SessionEvent<T>) => void
+	onStart?: (event: SessionEvent<T>) => void
+	onBeforeUpdate?: (event: SessionEvent<T>) => void
+	onUpdate?: (event: SessionEvent<T>) => void
+	onBeforeInterrupt?: (event: SessionEvent<T>) => void
+	onInterrupt?: (event: SessionEvent<T>) => void
+	onBeforeComplete?: (event: SessionEvent<T>) => void
+	onComplete?: (event: SessionEvent<T>) => void
+	onBeforeCancel?: (event: SessionEvent<T>) => void
+	onCancel?: (event: SessionEvent<T>) => void
+	onBeforeEnd?: (event: SessionEvent<T>) => void
+	onEnd?: (event: SessionEvent<T>) => void
+}
+
+/**
+ * A session info object that can be used to define event handlers for a session and provide other configuration info to a session.
+ * @public */
+export type SessionInfo<T extends object = object> = SessionEventHandlers & T
```

Comment:
should this be `SessionEventHandlers<T>`? Why are session event /session event handlers generic?